adapter/service/chargepoint/service.go

```diff
@@ -287,95 +271,24 @@ func (s *service) SupportedStates() []string {
 	return s.Service.Specification().PropertyStrings(PropertySupportedStates)
 }
 
-func (s *service) SupportsChargingModes() bool {
-	_, ok := s.controller.(AdjustableModeController)
-	if !ok {
-		return false
-	}
-
-	if len(s.Specification().PropertyStrings(PropertySupportedChargingModes)) == 0 {
-		return false
-	}
-
-	return true
-}
-
-func (s *service) SetChargingMode(mode string) error {
-	s.lock.Lock()
-	defer s.lock.Unlock()
-
-	if !s.SupportsChargingModes() {
-		return fmt.Errorf("%s: setting charging mode is unsupported", s.Name())
-	}
-
-	normalizedMode, ok := s.normalizeChargingMode(mode)
-	if !ok {
-		return fmt.Errorf("%s: unsupported charging mode: %s", s.Name(), mode)
-	}
-
-	controller, ok := s.controller.(AdjustableModeController)
-	if !ok {
-		return fmt.Errorf("%s: setting charging mode is unsupported", s.Name())
-	}
-
-	if err := controller.SetChargepointChargingMode(normalizedMode); err != nil {
-		return fmt.Errorf("%s: failed to set charging mode: %w", s.Name(), err)
-	}
-
-	return nil
-}
-
-func (s *service) SendChargingModeReport(force bool) (bool, error) {
-	s.lock.Lock()
-	defer s.lock.Unlock()
-
-	if !s.SupportsChargingModes() {
-		return false, fmt.Errorf("%s: charging mode report is unsupported", s.Name())
-	}
-
-	controller, ok := s.controller.(AdjustableModeController)
-	if !ok {
-		return false, fmt.Errorf("%s: charging mode report is unsupported", s.Name())
-	}
-
-	mode, err := controller.ChargepointChargingModeReport()
-	if err != nil {
-		return false, fmt.Errorf("%s: failed to get charging mode report: %w", s.Name(), err)
-	}
-
-	if !force && !s.reportingCache.ReportRequired(s.stateReportingStrategy, EvtChargingModeReport, "", mode) {
-		return false, nil
+// normalizeChargingMode normalizes provided charging mode. Returns true, when everything is fine.
+func (s *service) normalizeChargingMode(mode string) (string, error) {
+	if mode == "" {
```

Comment:
I wonder if we should send normal here if empty...

Reply:
As we discussed...